integration-tests/main/src/main/java/io/quarkus/it/config/MicroProfileConfigResource.java

```diff
@@ -38,6 +38,9 @@
     @Inject
     Config config;
 
+    @ConfigProperty(name = "greeting.empty")
+    String empty;
+
```

Comment:
would be good to also test the Optional case to be sure it gets an `Optional<String`> with an empty String

Reply:
actually, this is not correct according to the comment made above (`getOptionalValue returns an empty Optional when the string value is an empty string this should not be considered an error`).
I guess an Optional with an empty string must be injected in this case, not an empty optional.

Reply:
Actually we don't really control that part, the value comes from MP config (we just create the config source), so I'm not really sure what we should do.
Any ideas?

Reply:
@cescoffier the TCK passes, so I think this is safe. WDYT?

Reply:
@cescoffier ping.